ui/src/app/utils/navigation.tsx

```diff
@@ -0,0 +1,14 @@
+export const NavStore = {
+  navigate: undefined,
+  navigateByUrl: undefined
+};
+
+// NOTE: Because these are wired up directly to the router component,
+// all navigation done from here will effectively use absolute paths.
+export const navigate = (...args) => {
```

Comment:
Is it possible to just attach the `Router` here instead or are there issues? It may be needed in a few places e.g. for event subscription and it would keep type safety on the methods.